btk/draw_blends.py

```diff
@@ -456,7 +456,13 @@ class CosmosGenerator(DrawBlendsGenerator):
     def render_single(self, entry, filt, psf, survey, extra_data):
         """Returns the Galsim Image of an isolated galaxy."""
         galsim_catalog = self.catalog.get_galsim_catalog()
-        gal_flux = get_flux(entry["ref_mag"], filt, survey)
+
+        # get galaxy flux
+        try:
+            gal_flux = get_flux(entry["_".join([filt.name, survey.name])], filt, survey)
```

Comment:
```suggestion
            gal_flux = get_flux(entry[f"{filt.name}_{survey.name}"], filt, survey)
```
is probably more readable. Also, wouldn't reversing the survey and the filter make more sense ?

Reply:
I agree also with reversing the order of filter and survey (since filters live inside surveys)

Reply:
Yes indeed I just took back the column name from the underlying [EL COSMOS](http://cesam.lam.fr/aspic/files/elcosmos/readme_elcosmos.html) catalog but we can reverse.